python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -440,3 +442,78 @@ class DecorrelateALKernelMapReduceConfig(ImageMapReduceConfig):
         doc='A&L decorrelation subtask to run on each sub-image',
         target=DecorrelateALKernelMapperSubtask
     )
+
+
+class DecorrelateALKernelSpatialConfig(pexConfig.Config):
+    decorrelateConfig = pexConfig.ConfigField(
+        dtype=DecorrelateALKernelConfig,
+        doc='DecorrelateALKernel config to use when running on complete exposure (non spatially-varying)',
+    )
+
+    decorrelateMapReduceConfig = pexConfig.ConfigField(
+        dtype=DecorrelateALKernelMapReduceConfig,
+        doc='DecorrelateALKernelMapReduce config to use when running on each sub-image (spatially-varying)',
+    )
+
+    def setDefaults(self):
+        self.decorrelateMapReduceConfig.gridStepX = self.decorrelateMapReduceConfig.gridStepY = 19
+        self.decorrelateMapReduceConfig.gridSizeX = self.decorrelateMapReduceConfig.gridSizeY = 20
+        self.decorrelateMapReduceConfig.borderSizeX = self.decorrelateMapReduceConfig.borderSizeY = 6
+        self.decorrelateMapReduceConfig.reducerSubtask.reduceOperation = 'average'
+
+
+class DecorrelateALKernelSpatialTask(pipeBase.Task):
```

Comment:
I don't see the task documentation for doxygen; https://confluence.lsstcorp.org/display/DM/How+to+document+a+Task